docs/configuration/authentication.md

```diff
@@ -25,8 +25,32 @@ When using a GitHub personal access token, the username is arbitrary. GitHub doe
 logging in with password directly, although other hosts may. If a username is provided without
 credentials, you will be prompted to enter them.
 
+!!! important
+
+    If you use `uv add <url>` with a password or token in the URL as above, that credential will
+    _not_ be saved in `pyproject.toml` or `uv.lock`, because those files are often checked-in and
+    published. In that case, uv will probably fail with authentication errors the next time it tries
+    to fetch the new dependency, for example if you delete the lockfile or clone your project on a
+    new machine. It's better to configure Git to use your credential automatically as follows.
```

Comment:
...It seems weird to have a sidebar that then says "as follows"

Reply:
...should we be recommending these formats above if they're kinda broken like this?

Reply:
> ...should we be recommending these formats above if they're kinda broken like this?

We talked about this in Discord, those formats are totally valid for `uv pip install` and other local usage.